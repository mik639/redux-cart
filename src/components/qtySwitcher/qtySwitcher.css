.qty-switcher {
  display: flex;
}

.qty-switcher__value {
  display: inline-block;
  padding: 0 5px;
}