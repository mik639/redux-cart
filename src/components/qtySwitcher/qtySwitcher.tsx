import React from 'react';
import './qtySwitcher.css';

export interface PropsType {
  /**
   * Current value
   * @type {number}
   */
  value: number;
  /**
   * Minimum allowed value
   * @type {number}
   */
  min: number;
  /**
   * Handler for increase button
   */
  onIncrease: () => void;
  /**
   * Handler for decrease button
   */
  onDecrease: () => void;
}

export class QtySwitcher extends React.PureComponent<PropsType> {
  static defaultProps = {
    min: 1
  };

  /**
   * Handler for "decrease" button,
   * call onDecrease prop current value for than minimum allowed
   */
  decreaseHandler = () => {
    const { value, onDecrease, min } = this.props;
    if (value > min) onDecrease();
  };

  render() {
    const { onIncrease, value } = this.props;

    return (
      <div className="qty-switcher">
        <button onClick={this.decreaseHandler}>-</button>
        <span className="qty-switcher__value">{value}</span>
        <button onClick={onIncrease}>+</button>
      </div>
    );
  }
}
