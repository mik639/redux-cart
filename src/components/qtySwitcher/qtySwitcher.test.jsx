import * as React from 'react';
import renderer from 'react-test-renderer';
import { QtySwitcher } from './qtySwitcher';

describe('QtySwitcher', () => {
  let props;
  beforeEach(() => {
    props = {
      value: 3,
      onIncrease: jest.fn(),
      onDecrease: jest.fn()
    };
  });

  it('should renders correctly', () => {
    const tree = renderer.create(<QtySwitcher {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('decreaseHandler method', () => {
    it('should call onDecrease prop if current value more than minimum allowed', () => {
      const wrapper = renderer.create(
        <QtySwitcher {...props} value={2} min={1} />
      );

      expect(props.onDecrease).not.toHaveBeenCalled();
      wrapper.getInstance().decreaseHandler();
      expect(props.onDecrease).toHaveBeenCalled();
    });

    it('should NOT call onDecrease prop if current value is minimum allowed', () => {
      const wrapper = renderer.create(
        <QtySwitcher {...props} value={1} min={1} />
      );

      expect(props.onDecrease).not.toHaveBeenCalled();
      wrapper.getInstance().decreaseHandler();
      expect(props.onDecrease).not.toHaveBeenCalled();
    })
  });
});
