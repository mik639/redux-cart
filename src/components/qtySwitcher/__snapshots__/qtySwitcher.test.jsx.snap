// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QtySwitcher should renders correctly 1`] = `
<div
  className="qty-switcher"
>
  <button
    onClick={[Function]}
  >
    -
  </button>
  <span
    className="qty-switcher__value"
  >
    3
  </span>
  <button
    onClick={[MockFunction]}
  >
    +
  </button>
</div>
`;
